include/kokkos-utils/callbacks/EventBeginEndEventIdMatcher.hpp

```diff
@@ -21,7 +21,6 @@ struct EventBeginEndEventIdMatcher
     {
         if(matcher(event))
         {
-            if (event_id != invalid_event_id) Kokkos::abort("EventBeginEndEventIdMatcher cannot match a begin event twice.");
```

Comment:
Let's consider renaming this file and class `EventBeginEndIdMatcher` to avoid repeating the `Event` twice in the name.

Reply:
Also, we could have the class store the types of the begin and end events.

Reply:
Or `BeginEndEventIdMatcher`.

Reply:
I've tried to start with *Event* for matchers related to events. So I'm not in favor of `BeginEndEventIdMatcher`, but I agree it's weird repeating.

How about we just do `EventBeginEndIdMatcher`?

Reply:
Indeed, we can do `EventBeginEndIdMatcher`.